File: sdk/php/config/WebMediumConfig.php
<?php
/******************************************************************************
 *
 *  Copyright 2023 Paphus Solutions Inc.
 *
 *  Licensed under the Eclipse Public License, Version 1.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.eclipse.org/legal/epl-v10.html
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 ******************************************************************************/
require_once('Config.php');
require_once('./util/Utils.php');
abstract class WebMediumConfig extends Config
{
    public string $id;
    public string $name;
    public bool $isAdmin;
    public bool $isAdult;
    public bool $isPrivate;
    public bool $isHidden;
    public string $accessMode;
    public bool $isFlagged;
    public bool $isExternal;
    public bool $isPaphus;
    public bool $showAds = true;
    public string $forkAccessMode;
    public string $contentRating;
    public string $description;
    public string $details;
    public string $disclaimer;
    public string $website;
    public string $subdomain;
    public string $tags;
    public string $categories;
    public string $flaggedReason;
    public string $creator;
    public string $creationDate;
    public string $lastConnectedUser;
    public string $license;
    public string $avatar;
    public string $script;
    public string $graphic;
    public int $thumbsUp = 0;
    public int $thumbsDown = 0;
    public string $stars = "0";
    public string $connects;
    public string $dailyConnects;
    public string $weeklyConnects;
    public string $monthlyConnects;



    // abstract public function toXML() : String;
    abstract public function getType(): ?String;
    // abstract public function credentials(): WebMediumConfig;

    public function stats(): string
    {
        return "";
    }

    public function toString() : ?String {
		return $this->name;
	}

    public function getToken(): int
    {
        $token = 0;
        if (isset($this->token) && $this->token !== "") {
            $token = intval($this->token);
        }
        return $token;
    }
    public function writeXML(&$writer): void
    {
        $writer = $this->writeCredentails($writer);
        if (isset($this->id)) {
            $writer .= " id=\"" . $this->id . "\"";
        }
        if (isset($this->name)) {
            $writer .= " name=\"" . $this->name . "\"";
        }
        if (isset($this->isPrivate)) {
            $writer .= " isPrivate=\"true\"";
        }
        if (isset($this->isHidden)) {
            $writer .= " isHidden=\"true\"";
        }
        if (isset($this->accessMode)) {
            $writer .= " accessMode=\"" . $this->accessMode . "\"";
        }
        if (isset($this->contentRating)) {
            $writer .= " contentRating=\"" . $this->contentRating . "\"";
        }
        if (isset($this->forkAccessMode)) {
            $writer .= " forkAccessMode=\"" . $this->forkAccessMode . "\"";
        }
        if (isset($this->stars)) {
            $writer .= " stars=\"" . $this->stars . "\"";
        }
        if (isset($this->isAdult)) {
            $writer .= " isAdult=\"true\"";
        }
        if (isset($this->isFlagged)) {
            $writer .= " isFlagged=\"true\"";
        }
        if (isset($this->isExternal)) {
            $writer .= " isExternal=\"true\"";
        }
        if (isset($this->showAds)) {
            $writer .= " showAds=\"true\"";
        }
        $writer .= ">";
        if (isset($this->description)) {
            $writer .= "<description>";
            $writer .= Utils::escapeHTML($this->description);
            $writer .= "</description>";
        }
        if (isset($this->details)) {
            $writer .= "<details>";
            $writer .= Utils::escapeHTML($this->details);
            $writer .= "</details>";
        }
        if (isset($this->disclaimer)) {
            $writer .= "<disclaimer>";
            $writer .= Utils::escapeHTML($this->disclaimer);
            $writer .= "</disclaimer>";
        }
        if (isset($this->categories)) {
            $writer .= "<categories>";
            $writer .= $this->categories;
            $writer .= "</categories>";
        }
        if (isset($this->tags)) {
            $writer .= "<tags>";
            $writer .= $this->tags;
            $writer .= "</tags>";
        }
        if (isset($this->license)) {
            $writer .= "<license>";
            $writer .= $this->license;
            $writer .= "</license>";
        }
        if (isset($this->website)) {
            $writer .= "<website>";
            $writer .= $this->website;
            $writer .= "</website>";
        }
        if (isset($this->subdomain)) {
            $writer .= "<subdomain>";
            $writer .= $this->subdomain;
            $writer .= "</subdomain>";
        }
        if (isset($this->flaggedReason)) {
            $writer .= "<flaggedReason>";
            $writer .= Utils::escapeHTML($this->flaggedReason);
            $writer .= "</flaggedReason>";
        }
    }

    public function parseXML($xml): void
    {
        $xmlData = Utils::loadXML($xml);
        if($xmlData===false) {
            return;
        }
        $this->id = $xmlData->attributes()->id;
        $this->name = $xmlData->attributes()->name;
        $this->creationDate = $xmlData->attributes()->creationDate;
        $this->isPrivate = (bool) $xmlData->attributes()->isPrivate;
        $this->isHidden = (bool) $xmlData->attributes()->isHidden;
        $this->accessMode = $xmlData->attributes()->accessMode;
		$this->contentRating = $xmlData->attributes()->contentRating;
		$this->forkAccessMode = $xmlData->attributes()->forkAccessMode;
		$this->isAdmin =(bool) $xmlData->attributes()->isAdmin;
		$this->isAdult =(bool) $xmlData->attributes()->isAdult;
		$this->isFlagged = (bool)$xmlData->attributes()->isFlagged;
		$this->isExternal = (bool) $xmlData->attributes()->isExternal;
		$this->creator = $xmlData->attributes()->creator;
		$this->creationDate = $xmlData->attributes()->creationDate;
		$this->connects = $xmlData->attributes()->connects;
		$this->dailyConnects = $xmlData->attributes()->dailyConnects;
		$this->weeklyConnects = $xmlData->attributes()->weeklyConnects;
		$this->showAds = (bool)$xmlData->attributes()->showAds;
		$this->monthlyConnects = $xmlData->attributes()->monthlyConnects;
        //Thumbs up and down
        $this->thumbsUp = (int)$xmlData->attributes()->thumbsUp;
        $this->thumbsDown = (int)$xmlData->attributes()->thumbsDown;

        //Tag Names
        $this->description = $xmlData->description;
        $this->details = $xmlData->details;
        $this->disclaimer = $xmlData->disclaimer;
        $this->categories = $xmlData->categories;
        $this->tags = $xmlData->tags;
        $this->flaggedReason = $xmlData->flaggedReason;
        $this->lastConnectedUser = $xmlData->lastConnectedUser;
        $this->license = $xmlData->license;
        $this->website = $xmlData->website;
        $this->subdomain = $xmlData->subdomain;
        $this->avatar = $xmlData->avatar;
    }
}


?>

File: sdk/python/config/WebMediumConfig.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License")
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from abc import abstractmethod
from util.Utils import Writer
from util import Utils


class WebMediumConfig(Config):
    id:str
    name:str
    isAdmin:bool
    isAdult:bool
    isPrivate: bool
    isHidden: bool
    accessMode:str
    isFlagged: bool
    isExternal: bool
    isPaphus: bool
    showAds: bool
    forkAccessMode:str
    contentRating:str
    description:str
    details:str
    disclaimer:str
    website:str
    subdomain:str
    tags:str
    categories:str
    flaggedReason:str
    creator:str
    creationDate:str
    lastConnectedUser:str
    license:str
    avatar:str
    script:str
    graphic:str
    thumbsUp:int
    thumbsDown:int
    stars: str
    connects: str
    dailyConnects: str
    weeklyConnects: str
    monthlyConnects: str
    def __init__(self):
        super().__init__()
        self.id = None
        self.name = None
        self.isAdmin = False
        self.isAdult = False
        self.isPrivate = False
        self.isHidden = False
        self.accessMode = None
        self.isFlagged = False
        self.isExternal = False
        self.isPaphus = False
        self.showAds = True
        self.forkAccessMode = None
        self.contentRating = None
        self.description = None
        self.details = None
        self.disclaimer = None
        self.website = None
        self.subdomain = None
        self.tags = None
        self.categories = None
        self.flaggedReason = None
        self.creator = None
        self.creationDate = None
        self.lastConnectedUser = None
        self.license = None
        self.avatar = None
        self.script = None
        self.graphic = None
        self.thumbsUp = 0
        self.thumbsDown = 0
        self.stars = None
        self.connects = None
        self.dailyConnects = None
        self.weeklyConnects = None
        self.monthlyConnects = None

    @abstractmethod
    def getType(self) -> str:
        pass

    @abstractmethod
    def credentials(self):
        pass

    def stats(self) -> str:
        return ""

    def toString(self) -> str:
        return self.name

    def getToken(self) -> int:
        token = 0
        if(self.token != None):
            token = int(self.token)
        return token

    def writeXML(self, writer: Writer):
        self.writeCredentials(writer)
        if(self.id != None):
            writer.append(" id=\"" + self.id + "\"")

        if(self.name != None):
            writer.append(" name=\"" + self.name + "\"")

        if(self.isPrivate != None and self.isPrivate == True):
            writer.append(" isPrivate=\"true\"")

        if(self.isHidden != None and self.isHidden == True):
            writer.append(" isHidden=\"true\"")

        if(self.accessMode != None):
            writer.append(" accessMode=\"" + self.accessMode + "\"")

        if(self.contentRating != None):
            writer.append(" contentRating=\"" + self.contentRating + "\"")

        if(self.forkAccessMode != None):
            writer.append(" forkAccessMode=\"" + self.forkAccessMode + "\"")

        if(self.stars != None):
            writer.append(" stars=\"" + self.stars + "\"")

        if(self.isAdult != None and self.isAdult == True):
            writer.append(" isAdult=\"true\"")

        if(self.isFlagged != None and self.isFlagged == True):
            writer.append(" isFlagged=\"true\"")
        if(self.isExternal != None and self.isExternal == True):
            writer.append(" isExternal=\"true\"")
        if(self.showAds != None and self.showAds == True):
            writer.append(" showAds=\"true\"")
        if(self.isAdult != None and self.isAdult == True):
            writer.append(" isAdult=\"true\"")
            
        writer.append(">")
        if (self.description != None):
            writer.append("<description>")
            writer.append(Utils.escapeHTML(self.description))
            writer.append("</description>")

        if (self.details != None):
            writer.append("<details>")
            writer.append(Utils.escapeHTML(self.details))
            writer.append("</details>")

        if (self.disclaimer != None):
            writer.append("<disclaimer>")
            writer.append(Utils.escapeHTML(self.disclaimer))
            writer.append("</disclaimer>")

        if (self.categories != None):
            writer.append("<categories>")
            writer.append(Utils.escapeHTML(self.categories))
            writer.append("</categories>")

        if (self.tags != None):
            writer.append("<tags>")
            writer.append(Utils.escapeHTML(self.tags))
            writer.append("</tags>")

        if (self.license != None):
            writer.append("<license>")
            writer.append(Utils.escapeHTML(self.license))
            writer.append("</license>")

        if (self.subdomain != None):
            writer.append("<subdomain>")
            writer.append(Utils.escapeHTML(self.subdomain))
            writer.append("</subdomain>")

        if (self.flaggedReason != None):
            writer.append("<flaggedReason>")
            writer.append(Utils.escapeHTML(self.flaggedReason))
            writer.append("</flaggedReason>")

    def parseXML(self, xml):
        root = Utils.loadXML(xml)
        if(root == None):
            return
        print(root)
        self.id = root.attrib.get("id")
        self.name = root.attrib.get("name")
        self.creationDate = root.attrib.get("creationDate")
        self.isPrivate = root.attrib.get("isPrivate")
        self.isHidden = root.attrib.get("isHidden")
        self.accessMode = root.attrib.get("accessMode")
        self.contentRating = root.attrib.get("contentRating")
        self.forkAccessMode = root.attrib.get("forkAccessMode")
        self.isAdmin = root.attrib.get("isAdmin")
        self.isAdult = root.attrib.get("isAdult")
        self.isFlagged = root.attrib.get("isFlagged")
        self.isExternal = root.attrib.get("isExternal")
        self.creator = root.attrib.get("creator")
        self.creationDate = root.attrib.get("creationDate")
        self.connects = root.attrib.get("connects")
        self.dailyConnects = root.attrib.get("dailyConnects")
        self.weeklyConnects = root.attrib.get("weeklyConnects")
        self.showAds = root.attrib.get("showAds")
        self.monthlyConnects = root.attrib.get("monthlyConnects")
        self.thumbsUp = root.attrib.get("thumbsUp")
        self.thumbsDown = root.attrib.get("thumbsDown")

        # tag names
        self.description = root.find("description")
        self.details = root.find("details")
        self.disclaimer = root.find("disclaimer")
        self.categories = root.find("categories")
        self.tags = root.find("tags")
        self.flaggedReason = root.find("flaggedReason")
        self.lastConnectedUser = root.find("lastConnectedUser")
        self.license = root.find("license")
        self.website = root.find("website")
        self.subdomain = root.find("subdomain")
        self.avatar =root.find("avatar")
